components/ai_chat/resources/page/state/conversation_context.tsx

```diff
@@ -314,33 +319,25 @@ export function ConversationContextProvider(
   )
 
   const shouldShowLongConversationInfo = React.useMemo(() => {
+    // Get total number of characters in conversation history
     const chatHistoryCharTotal = context.conversationHistory.reduce(
       (charCount, curr) => charCount + curr.text.length,
       0
     )
 
-    // TODO(nullhook): make this more accurately based on the actual page content length
-    let totalCharLimit =
-      context.currentModel?.options.leoModelOptions !== undefined
-        ? context.currentModel.options.leoModelOptions
-            ?.longConversationWarningCharacterLimit
-        : loadTimeData.getInteger('customModelLongConversationCharLimit')
+    const options =
+      context.currentModel?.options.leoModelOptions ||
+      context.currentModel?.options.customModelOptions
 
+    let totalCharLimit = options?.longConversationWarningCharacterLimit ?? 0
     if (context.shouldSendPageContents) {
-      totalCharLimit +=
-        context.currentModel?.options.leoModelOptions !== undefined
-          ? context.currentModel.options.leoModelOptions?.maxPageContentLength
-          : loadTimeData.getInteger('customModelMaxPageContentLength')
+      totalCharLimit += options?.maxPageContentLength ?? 0
     }
 
-    if (
-      !hasDismissedLongConversationInfo &&
+    const conversationExceedsModelCapacity =
```

Comment:
I guess these names are a bit misleading because it's not the real capacity that is being exceeded but just the warning level